File: Two Sum/two_sum.java
import java.util.Hashtable;

class Solution {
	public int[] twoSum(int[] nums, int target) {
		Hashtable<Integer, Integer> table = new Hashtable<Integer, Integer>();
		table.put(Integer.valueOf(nums[0]), Integer.valueOf(0));
		
		Integer m;
		for (int i = 1; i < nums.length; i++) {
			try {	
				m = table.get(Integer.valueOf(target - nums[i]));
				int[] res = {i, m.intValue()};
				return res;
			} catch (Exception e) {
				table.put(Integer.valueOf(nums[i]), Integer.valueOf(i));
			}
		}
		return null;
	}
}

File: Two Sum/two_sum.jl
function twoSum(nums::Array{Int, 1}, target::Int)
	table = Dict(string(nums[1]) => 1)
	for i in range(2, length(nums))
		n = nums[i]
		m = get(table, string(target - n), false)
		if (m != false)
			return [i, m]
		else
			table[string(n)] = i 
		end
	end
end